### DISCUSSION TITLE
Interface can't produce traffict with vlan tagging

### DISCUSSION THREAD

**Discussion by xabudy** [rtbrick/bngblaster#230] [Q&A]
I have already installed bngblaster in my ubuntu vm. but when i try to configure to send dhcp ot pppoe traffict, it seems the traffic not tagged with vlan that configured in the json file. Please help to solve this problem?

**Comment by GIC-de** [comment]
Could you share Blaster configuration?

**Reply by xabudy** [reply]
Here i attach the json file sir. 
thanks in advance
[pppoe-test1.json](https://github.com/rtbrick/bngblaster/files/14321055/pppoe-test1.json)

**Reply by GIC-de** [reply]
How did you check that traffic is not tagged? It is important that the interface used is "blank" meaning no VLAN, IP, ... configured. You also need to consider that the outer VLAN header is often not captured in PCAP. Some interface drivers drop the outer VLAN header. The BNG Blaster tries to recover the VLAN from kernel headers but will not change the packets stored in the PCAP file.  

You should also capture with Blaster integrated capture (`bngblaster -P test.pcap ...`) instead of tcpdump, because of kernel bypass packets send by blast will not be seen in tcpdump. 

https://rtbrick.github.io/bngblaster/faq.html
